### PR TITLE
issue #2143 - documentation for the registry and how its used

### PR DESCRIPTION
**Pull request by lmsurpre**
Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

**Review by JohnTimm — APPROVED** [review]
LGTM

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 1]
```diff
@@ -691,20 +713,49 @@ To implement a persistence interceptor, complete the following steps:
 
 4.  Re-start the FHIR server.
 
-## 4.4 Resource validation
+## 4.4 Registry resources
+The IBM FHIR Server includes a dynamic registry of conformance resources.
+The registry pre-packages all [FHIR Definitions from the specification](https://www.hl7.org/fhir/R4/downloads.html)
+and uses a Java ServiceLoader to discover additional registry resource providers on the classpath.
+
+The server consults this registry for
```
Comment (tbieste) [inline docs/src/pages/guides/FHIRServerUsersGuide.md:721]: Add ":" at the end.

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 2]
```diff
@@ -691,20 +713,49 @@ To implement a persistence interceptor, complete the following steps:
...
+* StructureDefinition, ValueSet, and CodeSystem resources for resource validation.
+* SearchParameter and CompartmentDefinition resources for search.
+* ValueSet and CodeSystem resources for terminology operations.
+* And more.
+
+One common technique for extending FHIR with a set of conformance resources is to build or reference an [Implementation Guide](https://www.hl7.org/fhir/implementationguide.html).
+
+The IBM FHIR Server includes a [PackageRegistryResourceProvider](https://ibm.github.io/FHIR/guides/FHIRValidationGuide#making-profiles-available-to-the-fhir-registry-component-fhirregistry) for registeriing implementation guide resources.
```
Comment (tbieste) [inline docs/src/pages/guides/FHIRServerUsersGuide.md:729]: Misspelled "registeriing"

File: docs/src/pages/guides/FHIRServerUsersGuide.md (hunk 1 above)
Comment (lmsurpre) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:721]:
```suggestion
The server consults this registry for:
```

File: docs/src/pages/guides/FHIRServerUsersGuide.md (hunk 2 above)
Comment (lmsurpre) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:729]: My `i` key is getting sticky on this awful butterfly keyboard...I am making this typing mistake over and over...

Comment (lmsurpre) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:729]:
```suggestion
The IBM FHIR Server includes a [PackageRegistryResourceProvider](https://ibm.github.io/FHIR/guides/FHIRValidationGuide#making-profiles-available-to-the-fhir-registry-component-fhirregistry) for registering implementation guide resources.
```

**Pull request merged**